packages/dart_code/src/nodes.candy

```diff
@@ -156,13 +190,44 @@ impl DartBlock: DartBody
 
 // Types.
 
-trait DartType: DartExpression {}
+public trait DartType: DartExpression
+
+// TODO(JonasWanke): Move these to `DartType` when we support static properties in traits.
+static let dartBool = DartNamedType(DartIdentifier("bool", Some<String>(dartCoreUrl)), List.empty<DartType>())
+static fun dartComparable(t: DartType): DartType {
+  DartNamedType(DartIdentifier("Comparable", Some<String>(dartCoreUrl)), List.of1<DartType>(t))
+}
+static let dartDeprecated = DartNamedType(DartIdentifier("Deprecated", Some<String>(dartCoreUrl)), List.empty<DartType>())
+static let dartDouble = DartNamedType(DartIdentifier("double", Some<String>(dartCoreUrl)), List.empty<DartType>())
+static let dartDynamic = DartNamedType(DartIdentifier("dynamic", None<String>()), List.empty<DartType>())
+static let dartInt = DartNamedType(DartIdentifier("int", Some<String>(dartCoreUrl)), List.empty<DartType>())
+static fun dartIterable(entry: DartType): DartType {
+  DartNamedType(DartIdentifier("Iterable", Some<String>(dartCoreUrl)), List.of1<DartType>(entry))
+}
+static fun dartIterator(entry: DartType): DartType {
+  DartNamedType(DartIdentifier("Iterator", Some<String>(dartCoreUrl)), List.of1<DartType>(entry))
+}
+static fun dartList(entry: DartType): DartType {
+  DartNamedType(DartIdentifier("List", Some<String>(dartCoreUrl)), List.of1<DartType>(entry))
+}
+static fun dartMap(key: DartType, value: DartType): DartType {
+  DartNamedType(DartIdentifier("Map", Some<String>(dartCoreUrl)), List.of2<DartType>(key, value))
+}
+static let dartNull = DartNamedType(DartIdentifier("Null", Some<String>(dartCoreUrl)), List.empty<DartType>())
```

Comment:
We should make sure this doesn't get confused with the `null` constant (for which we should probably also have a constant here.

Reply:
For `null`, there's `DartNullLiteral()` already. Should there be a constant for that too? I'm not sure that's worth it, unless we make `DartNullLiteral` private.

Regarding the type constant: Should I rename it to `dartNullType`? That would differentiate it from other type constants, but it's clearer that it's a type. (This becomes irrelevant as soon as we can have static properties in traits, as we can move these to `DartType`.)

Reply:
Ah, didn't spot the `DartNullLiteral`. Then I'd leave this as-is, keep it consistent.